src/wasm/wasm-type.cpp

```diff
@@ -1297,6 +1300,8 @@ bool HeapType::isArray() const {
   }
 }
 
+bool HeapType::isString() const { return *this == HeapType::string; }
```

Comment:
This is the only predicate we have that is expressible as an equality comparison. Should we just use an equality comparison everywhere instead? We don't have `isNone()`, for example.

Reply:
I think it's clearer with `isString()`, but yeah, maybe it's excessive. I lean towards `isString()` for now as the proposal is still experimental, and it's easier to change (e.g. maybe we'll want to consider string iterators as strings as well..? I'm not sure).

Reply:
Yeah, fair enough that we might end up with multiple string types.